src/main/java/org/openrewrite/maven/AbstractRewriteBaseRunMojo.java

```diff
@@ -54,6 +62,10 @@ public abstract class AbstractRewriteBaseRunMojo extends AbstractRewriteMojo {
 
     @Parameter(property = "rewrite.exportDatatables", defaultValue = "false")
     protected boolean exportDatatables;
+    @Parameter(property = "rewrite.options")
+    protected Map<String, String> options;
+    @Parameter(property = "rewrite.flatOptions")
+    protected String flatOptions;
```

Comment:
I'm leaning towards only having one or the other, as we're mostly targeting one-off single recipe runs here, I think a flow aimed exclusively at command line argument handling is best. I wouldn't expect or suggest folks configure single recipes in their `pom.xml`, as the better fit there would be a `rewrite.yml` file containing those same options.

Reply:
The more a use rewrite maven plugin, the more I think it needs to be slightly different...

For this, `flatOptions` is a must have for command line, as `options` don't work from command line due to maven not parsing complex types from command line argument.

But, I think `recipes` should be different, like this:


```
  <activeRecipes>
    <activeRecipe>
      <name>org.openrewrite.java.testing.hamcrest.MigrateHamcrestToAssertJ</name>
      <exclusions>
        <exclusion>**/pom.xml</exclusion>
      </exclusions>
    </activeRecipe>
    <activeRecipe>
      <name>org.openrewrite.java.testing.hamcrest.HamcrestMatcherToAssertJ</name>
      <options>
        <option>
          <matcher>fooBar</matcher>
          <assertion>isFoo</assertion>
          <argumentType>my.Bar</argumentType>
        </option>
      </options>
    </activeRecipe>
    <activeRecipe>
      <name>org.openrewrite.java.testing.assertj.JUnitToAssertj</name>
    </activeRecipe>
    <activeRecipe>
      <name>org.openrewrite.java.testing.assertj.Assertj</name>
    </activeRecipe>
    <activeRecipe>
      <name>org.openrewrite.java.migrate.UpgradeToJava21</name>
      <exclusions>
        <exclusion>**/src/main/java/**</exclusion>
        <exclusion>**/pom.xml</exclusion>
      </exclusions>
    </activeRecipe>
  </activeRecipes>
```

This would be very useful for complex automatically review PRs with OpenRewrite recipes.

Reply:
I'd hold off any such changes; we have other ways to support such a workflow if needed, for instance using profiles, or add your recipes separated by `---` to a `rewrite.yml` in the root of your project. That last option has the benefit of also being supported in IntelliJ Ultimate by default.

The more complicated configuration options seem like they would be challenging to maintain going forward. We'd really want to push the configuration into options supported by Maven, Gradle, Moderne CLI, JetBrains' OpenRewrite plugin for IntelliJ, Moderne Platform, ... Having a structure that works for all those options is then preferred over something unique to the Maven plugin.